### PR TITLE
[StorageSync] Address self-review comments

### PR DESCRIPTION
**Pull request by fengzhou-msft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs [hunk 1]
```diff
@@ -170,22 +170,61 @@ public partial class ServerEndpointResource : Azure.ResourceManager.ArmResource
...
-        public static System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.StorageSync.Models.CheckNameAvailabilityResult>> CheckNameAvailabilityStorageSyncServiceAsync(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, string locationName, Azure.ResourceManager.StorageSync.Models.CheckNameAvailabilityContent content, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static Azure.Response<Azure.ResourceManager.StorageSync.Models.StorageSyncNameAvailabilityResult> CheckNameAvailabilityStorageSyncService(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, string locationName, Azure.ResourceManager.StorageSync.Models.StorageSyncNameAvailabilityContent content, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (ArthurMa1978) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs:173]: Then func name should be `CheckStorageSyncNameAvailability`

File: sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs [hunk 2]
```diff
@@ -267,108 +306,62 @@ public partial class StorageSyncServiceResource : Azure.ResourceManager.ArmResou
...
+        System.Collections.Generic.IEnumerator<Azure.ResourceManager.StorageSync.StorageSyncWorkflowResource> System.Collections.Generic.IEnumerable<Azure.ResourceManager.StorageSync.StorageSyncWorkflowResource>.GetEnumerator() { throw null; }
         System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
     }
-    public partial class WorkflowData : Azure.ResourceManager.Models.ResourceData
+    public partial class StorageSyncWorkflowData : Azure.ResourceManager.Models.ResourceData
     {
-        public WorkflowData() { }
+        public StorageSyncWorkflowData() { }
         public string CommandName { get { throw null; } }
         public System.DateTimeOffset? CreatedTimestamp { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs:342]: CreatedOn

**Review by ArthurMa1978 — APPROVED** [review]

**Comment by ArthurMa1978** [comment]
> **API change check**
> 
> APIView has identified API level changes in this PR and created following API reviews.
> 
> [Azure.ResourceManager.StorageSync](https://apiview.dev/Assemblies/Review/a9bae67bd5aa4456bf392feeebeab4d0)

some comments in the apiview

**Comment by fengzhou-msft** [comment]
> > **API change check**
> > APIView has identified API level changes in this PR and created following API reviews.
> > [Azure.ResourceManager.StorageSync](https://apiview.dev/Assemblies/Review/a9bae67bd5aa4456bf392feeebeab4d0)
> 
> some comments in the apiview

resolved.

**Pull request merged**